### DISCUSSION TITLE
Fitment of Sidebar & Sidetabs & Suggestion For Tab Group Addition

### DISCUSSION THREAD

**Discussion by TooSpooky** [Floorp-Projects/Floorp#129] [Q&A]
Hello,

I'm genuinely liking the additions you have made so far in Floorp. I've been using Librewolf with a sidetab CSS but having it stock is really nice along with a sidebar.

Although I'm having a few oddities with fitment, shown here. 
<img width="36" alt="floorp_bf2WDxVlQN" src="https://user-images.githubusercontent.com/21211292/193982201-42aae6f4-860f-4064-b5d1-ecd1a74a64b5.png">
The sidebar has excess space and gives me a sidescroll bar instead of fully fitting the side of the window.
<img width="38" alt="floorp_QYFpc7zZxt" src="https://user-images.githubusercontent.com/21211292/193982295-db57ba12-c899-4303-916f-b935ed20a8c9.png">
And the sidetab with treestyle tabs fitment of the tab icons are a little wonky, if possible it'd be nice to be able to center them and keep them in-line. 
Same with the close button encroaching on the tab name.
<img width="27" alt="floorp_H2QK6cTorh" src="https://user-images.githubusercontent.com/21211292/193982438-ef90d7b1-eb64-4881-a25d-3e7636a0c8f1.png">

For reference here's how it looks using my Librewolf setup (It utilizes tab center reborn) for fitment.
<img width="168" alt="librewolf_hzA73Udl7V" src="https://user-images.githubusercontent.com/21211292/193982511-b6993533-6349-4255-9992-4e0a24380596.png">

Thanks for your time and I appreciate it!

As an aside I've been using panorama tab groups to somewhat emulate how the Bonbon browser presents tabs as I definitely prefer that method. Would be curious if it would be possible in the future to integrate a tab group feature into the main UI. Below or above the search bar.
<img width="751" alt="BonBon_cAsbxhcc8l" src="https://user-images.githubusercontent.com/21211292/193983114-46e3c6c5-7679-4cfe-beb2-9d8543a96397.png"> 
(bonbon tabs)
with the option to open up the full panorama tab group interface to select a tab group as well. Unsure the feasibility but thought i'd throw out the suggestion!

**Comment by Ryosuke-Asano** [comment]
>The sidebar has excess space and gives me a sidescroll bar instead of fully fitting the side of the window.

It will be fixed at 10.6.1


> with the option to open up the full panorama tab group interface to select a tab group as well. Unsure the feasibility but thought i'd throw out the suggestion!

Try to install PanoramaView addon https://addons.mozilla.org/ja/firefox/search/?q=panorama


I'm currently at school, so I'll reply to you later about TST. Please let me know if you have more information such as how to customize it with Librewolf.

**Comment by Ryosuke-Asano** [comment]
I forgot to tell you. Thank you for using Floorp

**Comment by TooSpooky** [comment]
> > The sidebar has excess space and gives me a sidescroll bar instead of fully fitting the side of the window.
> 
> It will be fixed at 10.6.1
> 
> > with the option to open up the full panorama tab group interface to select a tab group as well. Unsure the feasibility but thought i'd throw out the suggestion!
> 
> Try to install PanoramaView addon https://addons.mozilla.org/ja/firefox/search/?q=panorama
> 
> I'm currently at school, so I'll reply to you later about TST. Please let me know if you have more information such as how to customize it with Librewolf.

Here’s the CSS I use for Librewolf.
https://github.com/ranmaru22/firefox-vertical-tabs

Good to know the sidebar will be fixed, thanks.

And I currently use a panorama extension, was just curious if in the future something like that could be integrated so you could switch between tab groups without going to a different UI. I.E. tab groups in the place and form of where horizontal tabs used to be if you want to change between them quickly. 

Thanks for the quick reply and take care!

**Comment by Ryosuke-Asano** [comment]
So you're talking about quick-switching tab groups in the native implementation. It's in the research stage, but not yet practical.

**Comment by TooSpooky** [comment]
Yes,

Ah alright, good to hear it’s in the research stage and totally understand. Thanks.

**Comment by Ryosuke-Asano** [comment]
If it does, I'll warn you that it will be a feature similar to Opera's workspaces.

**Comment by TooSpooky** [comment]
Also one or two more things,

When adding gmail.com to the sidebar, when you open it and click sign in it sends the login screen to the main browser tabs, was just curious if that was intended or not. Easy fix to just add the inbox url instead though. 

And have you thought of adding a notes feature to the sidebar? 

Thanks again, sorry for all the questions haha. 

Also good to know, not actually familiar with their implementation. Interesting.

**Comment by Ryosuke-Asano** [comment]
> When adding gmail.com to the sidebar, when you open it and click sign in it sends the login screen to the main browser tabs, was just curious if that was intended or not. Easy fix to just add the inbox url instead though.

This specification is not modified (in the sense that it cannot be) as it conforms to the Firefox specification


> And have you thought of adding a notes feature to the sidebar?

I love Vivaldi so the possibility of it being implemented is great

**Comment by TooSpooky** [comment]
Ah sweet, think it’d be a good addition. 

And ah gotcha on the sidebar, not a problem. 

Thanks again for your quick replies. Take care!

**Comment by TooSpooky** [comment]
Hey haha,

Forgot one question. 
Would it be possible to add a "Open in sidebar" feature to the right click menu when right clicking a link, image, or even potentially a tab? 
Since you can add custom url's and open them in sidebar it makes me wonder if that's possible since it's somewhat similar.

Thanks!

**Comment by Ryosuke-Asano** [comment]
It's possible, but you'll have to rewrite the code. Consider. We are currently rewriting the source code.

**Comment by Ryosuke-Asano** [comment]
Hi, We will add memo(notes) function & native tabbar function!

![image](https://user-images.githubusercontent.com/73892113/227713871-8f158428-2f31-43c0-a236-2fdf9682e18e.png)
![image](https://user-images.githubusercontent.com/73892113/227713872-9c442cd9-821a-486e-89f3-b2c0e2f73ec6.png)

**Comment by TooSpooky** [comment]
Hey, looks great! Been dailying floorp for awhile and it keeps improving! 

Looking forward to giving those a shot for sure.

**Comment by R0M4N0F5K7** [comment]
Hi, continuing the discussion on the fitment of the vertical tabs: It bothers me too that they are cut off depending on the tree hierarchy indent if the tab bar is closed. Is there a way to set the tab tree max indent to 0 when the bar is minimized so that the icons are nicely centered in that state? There must be a way as they definitely expand further when sidetabbar is opened...
The closest thing I found was the base or max indent setting in the TST dev settings which limits how much they separate when extending the bar, but then they also don't expand anymore when the sidetabs are expanded...

**Reply by Ryosuke-Asano** [reply]
TST will be removed from Floorp. As noted above, vertical tabs are replaced by Floorp's native implementation.  Improvements to TST will no longer be implemented

**Reply by R0M4N0F5K7** [reply]
Makes sense. With the native implementation all tabs will be on the same level anyway, right? So it just would be like TST with hierarchy level set to 0 anyway.
I saw that you said it will be implemented in version 11, but no mention of when 11 will release. Any estimates?

**Reply by Ryosuke-Asano** [reply]
Floorp based on ESR 115 is Floorp 11. The stable release will come at July 4 (maybe)

**Comment by Ryosuke-Asano** [comment]
More news!

Floorp will have Workspaces (tab group). at Floorp 11

![image](https://github.com/Floorp-Projects/Floorp/assets/73892113/a8ec499b-3b6b-4064-9a3e-627351c2687c)
